File: src/calculatorCli.js
function calculate(){
    while(true){
        console.log('0. exit\n1. add\n2. subtract\n3. multiply\n4. divide\n5. factorial\n6. power \n7. log\n8. square root \n9. sin\n10. cos\n11. tan')
        let userInput = prompt('Enter the number of the operation: ').toLowerCase();
        if(userInput === '0'){
            break;
        }
        
        switch(userInput){
            case '1':
                performCalculation(addition);
                break;
            case '2':
                performCalculation(subtraction);
                break;
            case '3':
                performCalculation(multiplication);
                break;
            case '4':
                performCalculation(division);
                break;
            case '5':
                factorial();
                break;
            case '6':
                performCalculation(power);
                break;
            case '7':
                logarithm();
                break;
            case '8':
                squareRoot();
                break;
            case '9':
                sin();
                break;
            case '10':
                cos();
                break;
            case '11':
                tan();
                break;
            default:
                console.log('Invalid operation. Please check your operation.\n')
        }
    }
}

function performCalculation(operation){
    const input1 = parseFloat(prompt('Enter first input: '));
    const input2 = parseFloat(prompt('Enter second input: '));
    
    const result = operation(input1,input2);
    console.log(`Result: ${result}\n`);
}


function addition(a, b){
    return a + b;
}

function subtraction(a, b){
    return a - b;
}

function multiplication(a, b){
    return a * b;
}

function division(a, b){
    if(b === 0){
        return 'Infinity';
    }
    return a / b
}

function factorial(){
    const input1 = parseFloat(prompt('Enter first input: '));
    
    let result = 1;
    if(input1 === 0 || input1 === 1){
        console.log('Result: ' + 1  + '\n');
    } else {
         for(let i = 2;i <= input1; i++){
            result *= i;
        }
        console.log(`Result: ${result} \n`);
    }
}

function power(a, b){
    return Math.pow(a,b);
}

function logarithm(){
    const input1 = parseFloat(prompt('Enter first input: '));
    
    let result = Math.log10(input1);
    console.log(`Result: ${result} \n`);
}

function squareRoot(){
    const input1 = parseFloat(prompt('Enter first input: '));
    
    let result = Math.sqrt(input1);
    console.log(`Result: ${result} \n`);
}

function sin(){
    const input1 = parseFloat(prompt('Enter first input: '));
    
    let result = Math.sin(input1);
    console.log(`Result: ${result} \n`);
}

function cos(){
    const input1 = parseFloat(prompt('Enter first input: '));
    
    let result = Math.cos(input1);
    console.log(`Result: ${result} \n`);
}

function tan(){
    const input1 = parseFloat(prompt('Enter first input: '));
    
    let result = Math.tan(input1);
    console.log(`Result: ${result} \n`); 
}


export default calculate;

File: src/calculatorCli.py
import math

def calculate():
    while True:
        print('\n1. add         6. power         11. tan' +      
              '\n2. subtract    7. log10          0. exit' + 
              '\n3. multiply    8. square root' + 
              '\n4. divide      9. sin' +
              '\n5. factorial  10. cos' )
        user_input = input('Select the number of the operation: ')
        
        if user_input == '0':
            break
        
        match user_input:
            case '1':
                perform_calculation(addition)
            case '2':
                perform_calculation(subtraction)
            case '3':
                perform_calculation(multiplication)
            case '4':
                perform_calculation(division)
            case '5':
                factorial()
            case '6':
                perform_calculation(power)
            case '7':
                logarithm()
            case '8':
                square_root()
            case '9':
                sin()
            case '10':
                cos()
            case '11':
                tan()
            case default:
                print('\nInvalid operation. Please check your operation.')

def perform_calculation(operation):
    try:
        input_1 = float(input('Enter first input: '))
        input_2 = float(input('Enter second input: '))
        
        result = operation(input_1,input_2)
        print(f"\nResult: {result}")
    except ValueError:
        print(f"\nInvalid input: Values must be number.")

def addition(a, b):
    return a + b
    
def subtraction(a, b):
    return a - b

def multiplication(a, b):
    return a * b

def division(a, b):
    if b == 0:
        return 'Infinity'
    
    return a / b

def factorial():
    try:
        input1 = int(input('Enter first input: '))
        print(f"\nResult: {math.factorial(input1)}")
    except ValueError:
        print('\nInvalid input: Values must not be negative or string.')

def power(a, b):
    return math.pow(a,b)


def logarithm():
    try:
        input1 = int(input('Enter first input: '))
        print(f"\nResult: {math.log10(input1)}")
    except ValueError:
        print('\nInvalid input: Values must not be negative or string.')


def square_root():
    try:
        input1 = float(input('Enter first input: '))
        print(f"\nResult: {math.sqrt(input1)}")
    except ValueError:
        print('\nInvalid input: Values must not be negative or string.')

def sin():
    try:
        input1 = float(input('Enter first input: '))
        print(f"\nResult: {math.sin(input1)}")
    except ValueError:
        print('\nInvalid input: Values must be a number.')

def cos():
    try:
        input1 = float(input('Enter first input: '))
        print(f"\nResult: {math.cos(input1)}")
    except ValueError:
        print('\nInvalid input: Values must be a number.')

def tan():
    try:
        input1 = float(input('Enter first input: '))
        print(f"\nResult: {math.tan(input1)}")
    except ValueError:
        print('\nInvalid input: Values must be a number.')


calculate()